### PR TITLE
Finish adding BitIndex 3

### PR DESCRIPTION
**Pull request by joshua-s**
This PR adds the remainder of the endpoints for BitIndex API v3.

### CODE REVIEW

**Comment by attilaaf** [comment]
@joshua-s I had a problem with STN on BitIndex (I forgot to disable cashaddr after reloading and hence why it was down).

My apologies for that, I just fixed and tested it works now.

STN is now running properly.

Thank you

**Comment by joshua-s** [comment]
@attilaaf thanks. Are you a developer of BitIndex? If so, I have some questions about the API.

**Comment by attilaaf** [comment]
@joshua-s Yes, I am. Let me know how I can help.  I'm also in Atlantis if you are there too

**Comment by AustEcon** [comment]
This is awesome. Thanks!
Do you need me to merge these changes quickly? 
Or are you operational for hackathon? I only have 10 hours left for hackathon now then work on Monday....

**Comment by joshua-s** [comment]
@AustEcon no, you can leave these until after the hackathon. Good luck on your stuff!

**Comment by joshua-s** [comment]
@AustEcon @attilaaf what do you think about splitting out the BitIndex integration into a new python package?

This would do several good things:
- It would be easily discoverable for people searching for a BitIndex integration
- It would allow people who just want to consume BitIndex to do so without importing the whole of bitsv
- It would reduce the scope of bitsv and make maintenance and testing simpler

**Comment by attilaaf** [comment]
@joshua-s @AustEcon I will defer to you guys on what you prefer to do with your project.  

I'm here to help answer any questions and help fix any BitIndex bugs -- either way I'm happy there is Python support for BitIndex via either path :)

**Comment by AustEcon** [comment]
Awesome stuff. I have tested all except for the xpub related stuff. I am happy to merge this now. I don't like to keep PRs floating if I can help it.

A couple of simple things I noticed that I want to ask about:

1) Was it intentional to remove the default parameters of:
- query="getInfo" as default for `bix3.get_network_status('getInfo')`  ?
- from_index=0 and to_index=0 as defaults for `bix3.get_transactions(my_key.address, from_index=0, to_index=1)  `

For `bix3.get_utxos(my_key.address, sort='value:desc')`
- I think it would be better to have a default value here as e.g. 'desc'. 
- And I also think the syntax would be improved by converting from "value:desc" and "value:asc" --> simply "desc" and "asc".

Regarding refactoring out BitIndex as a standalone repo... I'm not opposed to the idea and I would hate for bitsv to become any kind of bottle neck for the breakneck pace of bitcoin sv developments... (As much as I love people using bitsv just to get at the juicy juicy network apis 😆😃). It's a big change so would want to run it by @teran-mckinney and a few others to see how they all feel. For now I'm pretty happy keeping it all in-house and integrating it into the travis CI testing modules etc. But if you want to go and build it as a standalone, then I will take your lead and import the BitIndex api that way. I think you make some great points.

Really appreciate all the work you've done here.

**Comment by joshua-s** [comment]
@AustEcon thanks!

With regards to your questions/comments:

> Was it intentional to remove the default parameters

I removed those defaults because they were not consistent with the rest of the integration. This client was implemented as a direct interface to the BitIndex API v3, no magic included. The only parameters I specified defaults for are the ones that had defaults specified in BitIndex's documentation. I have noticed a bug though: the optional parameters are not set properly (they should be initialized to `None`).

> I think it would be better to have a default value here as e.g. 'desc'. And I also think the syntax would be improved by converting from "value:desc" and "value:asc" --> simply "desc" and "asc".

That's a good idea and will make it more dev-friendly. We could probably either:
1. Accept two optional parameters: `sort` and `sort_direction`. `sort_direction` would default to `'desc'`.
2. Accept a tuple for the `sort` parameter, with the value as the first item and the direction as the second. The second could be omitted and would default to `'desc'`.

> Regarding refactoring out BitIndex as a standalone repo... I'm not opposed to the idea and I would hate for bitsv to become any kind of bottle neck for the breakneck pace of bitcoin sv developments... (As much as I love people using bitsv just to get at the juicy juicy network apis 😆😃).

Gotta love those network APIs! I do agree that it is very nice that bitsv comes with batteries included: it is a great one-stop shop for development on bitcoin (sv).

I might still create a standalone package, but we will see.

**Comment by teran-mckinney** [comment]
For understanding if bitindex should be standalone or not, could you tell me what it is and does? I'm not deeply involved with SV and am pretty behind that way. I don't even know what STN is. If you can give me some basic links to read up I can give you more of an idea.

I am generally a fan of KISS and modularity.

**Comment by attilaaf** [comment]
@teran-mckinney I'll jump in real quick (creator of BitIndex here).

BitIndex is a Bitcoin SV blockchain API -- it provides REST Insight API compatibility.  

In other words it allow you to query utxo's and transactions and also broadcast transactions in a standard way.

It's basically Bitpay's Insight-API built for Bitcoin SV.

Hope this helps...

File: bitsv/network/services/bitindex3.py [hunk 1]
```diff
@@ -61,71 +65,291 @@ def get_balance(self, address):
...
+        :param address: Address or xpub key to track and monitor
+        """
+        r = requests.put(
+            f'https://api.bitindex.network/api/v3/{self.network}/webhook/monitored_addrs',
+            data=json.dumps({'addr': address}),
+            headers=self.authorized_headers,
+        )
+        return r.json()
+
+    def webhook_sample_callback(
```
Comment (attilaaf) [inline bitsv/network/services/bitindex3.py]: @AustEcon This is not a 'real' API call.

I should have been more explicit in the docs.  This is a SAMPLE of what the user will receive once they have register an endpoint to callback to their URL.

**Comment by teran-mckinney** [comment]
Being like Insight doesn't sound any different than what we normally have. I guess ideally you might have a Python 3 library with all of the API calls in a more Pythonic form and we call those, but it seems simple enough to have the support here in that case.

I guess ideally there would be an upstream Insight binding that we tap into, BitIndex, etc. But since the functionality is around payments, that seems normal to me.

File: bitsv/network/services/bitindex3.py (hunk 1 above)
Comment (joshua-s) [reply bitsv/network/services/bitindex3.py]: @attilaaf thanks for the clarification. This is removed in https://github.com/AustEcon/bitsv/pull/23

**Pull request merged**